### PR TITLE
feat: rename packages

### PR DESCRIPTION
**Pull request by dhayab**
**Summary**

This PR removes the following packages from the monorepo:
- react-instantsearch
- react-instantsearch-dom
- react-instantsearch-dom-maps
- react-instantsearch-native

It then renames the packages in this fashion:
- react-instantsearch-hooks-web → react-instantsearch
- react-instantsearch-hooks → react-instantsearch-core
- react-instantsearch-hooks-router-nextjs → react-instantsearch-router-nextjs
- react-instantsearch-hooks-server → all files included in react-instantsearch-core/server

Misc:
- react examples are replaced with those from react-hooks
- configuration and ci is updated

FX-2338

> **Note**  
> It will probably be easier to review commit by commit, and ignore the first one ("move packages") as it just shuffles around a bunch of files.

### CODE REVIEW

File: README.md
Comment (aymeric-giraudet) [inline README.md:1]: In "Contributing" bullet point 4, it still mentions `react-instantsearch-hooks`

**Review by aymeric-giraudet — COMMENTED** [review]
There are some mentions of RISH in CISA (or maybe we can wait until we remove the templates) and instantsearch.css's README files too

**Comment by dhayab** [comment]
@aymeric-giraudet @Haroenv I've made changes according to your findings, thanks!

Regarding CISA, it will have to be updated after we do the v7 release, or else the pipeline will fail.

I re-removed the warning, it might be related to an issue in a rebase, but I didn't have any conflict on my end, weird.

**Review by Haroenv — APPROVED** [review]
- examples + website seem to be correctly (completely) removed (including storybook)
- packages updated, build seems to work
- removed no longer needed packages (some more I'm doing in a followup pr)

**Review by aymeric-giraudet — APPROVED** [review]

**Pull request merged**